infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LogReplicationContext.java

```diff
@@ -68,13 +66,14 @@ public void setIsLeader(boolean newValue) {
     }
 
     /**
-     * This method will be invoked when it is needed to check if registry has new entries, to get the up-to-date
-     * LogReplicationConfig, which mainly includes streams to replicate and data streams to tags map. Please note
-     * that this method is synchronized because LogReplicationContext is shared across sessions so each session
-     * will have its own threads to access it.
+     * This method will be invoked when it is needed to get the up-to-date LogReplicationConfig by checking registry
+     * table or other config related tables.
+     *
+     * @param session LogReplicationSession to refresh the config.
+     * @param registryTableOnly True if only registry table is used for config generation or update.
      */
-    public synchronized void refresh() {
-        this.configManager.getUpdatedConfig();
+    public void refreshConfig(LogReplicationSession session, boolean registryTableOnly) {
```

Comment:
QQ: why synchronization is no longer needed here?

Reply:
In previous implementation all the calls to this method (from all the sessions) are totally serialized, and under the hood if one session found that registry table updated, it will update config for all the sessions. But now the case if for LOGICAL_GROUP sessions, they have to update their config in their own pace (otherwise they could miss the required snapshot sync). That's why this method is parameterized by session now, and under the hood "syncWithRegistryTable" and "syncWithClientConfigTable" are still synchronized.